File: selenium-java-hybrid/src/main/java/pageObjects/DashboardPage.java
package pageObjects;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

/**
 * Page Object for Dashboard Page
 * Project: 360GlobalNet
 */
public class DashboardPage {
    private WebDriver driver;
    private WebDriverWait wait;

    // Locators
    private By dashboardHeader = By.xpath("//h6[contains(text(),'Dashboard')]");
    private By dashboardContainer = By.className("oxd-topbar-header-breadcrumb");

    public DashboardPage(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    public boolean isDashboardDisplayed() {
        try {
            WebElement dashboard = wait.until(ExpectedConditions.visibilityOfElementLocated(dashboardHeader));
            return dashboard.isDisplayed();
        } catch (Exception e) {
            return false;
        }
    }

    public String getDashboardHeaderText() {
        WebElement header = wait.until(ExpectedConditions.visibilityOfElementLocated(dashboardHeader));
        return header.getText();
    }
}

File: pip/pageObjects/DashboardPage.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class DashboardPage:
    """
    Page Object for Dashboard Page
    """

    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 10)

    # Locators
    DASHBOARD_HEADER = (By.XPATH, "//h1[contains(text(),'Dashboard')]") 
    DASHBOARD_CONTAINER = (By.CLASS_NAME, "dashboard")
    USER_PROFILE = (By.ID, "user-profile")

    def is_dashboard_displayed(self):
        """
        Verify if dashboard page is displayed
        """
        try:
            self.wait.until(EC.presence_of_element_located(self.DASHBOARD_CONTAINER))
            return True
        except:
            return False

    def get_dashboard_title(self):
        """
        Get the dashboard page title
        """
        return self.driver.title
